build/commands/scripts/sync.js

```diff
@@ -100,6 +100,9 @@ function buildDefaultGClientConfig() {
   if (config.targetOS) {
     out += toGClientConfigItem('target_os', [config.targetOS], false)
   }
+  if (config.targetArch) {
+    out += toGClientConfigItem('target_cpu', [config.targetArch], false)
```

Comment:
Why is this needed? I think this is going to cause problems for anyone who builds more than one arch in the same checkout

Reply:
Linux arm64 builds don't work without.

Reply:
What's the alternative? Below a current log with this change applied.

```
19:08:33  ________ running 'python3 src/build/linux/sysroot_scripts/install-sysroot.py --arch=arm64' in '/home/ubuntu/workspace/test-brave-browser-build-linux-arm64-nightly'
19:08:33  Installing Debian bullseye arm64 root image: /home/ubuntu/workspace/test-brave-browser-build-linux-arm64-nightly/src/build/linux/debian_bullseye_arm64-sysroot
19:08:33  Downloading https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/f00ece500aef0ff2a431a07de48bd3e1aa6d1caf/debian_bullseye_arm64_sysroot.tar.xz
19:08:41  ________ running 'python3 src/build/linux/sysroot_scripts/install-sysroot.py --arch=x86' in '/home/ubuntu/workspace/test-brave-browser-build-linux-arm64-nightly'
19:08:41  Installing Debian bullseye i386 root image: /home/ubuntu/workspace/test-brave-browser-build-linux-arm64-nightly/src/build/linux/debian_bullseye_i386-sysroot
19:08:41  Downloading https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/07bb72368889fea19a1f123aa11528f0d792121b/debian_bullseye_i386_sysroot.tar.xz
19:08:48  ________ running 'python3 src/build/linux/sysroot_scripts/install-sysroot.py --arch=x64' in '/home/ubuntu/workspace/test-brave-browser-build-linux-arm64-nightly'
19:08:48  Installing Debian bullseye amd64 root image: /home/ubuntu/workspace/test-brave-browser-build-linux-arm64-nightly/src/build/linux/debian_bullseye_amd64-sysroot
19:08:48  Downloading https://commondatastorage.googleapis.com/chrome-linux-sysroot/toolchain/cb4fa34f1faddafb72cace35faf62a611f2ca7c9/debian_bullseye_amd64_sysroot.tar.xz
```

Reply:
On Linux, `gclient runhooks` by default doesn't run hooks for `arm64`. This leads to a missing sysroot directory when building for that architecture. The change makes sure that `gclient` does include the hooks for `arm64`.

Regarding problems for people building more than one arch: From `gclient.py`:

```
# Specifying a target CPU
#   To specify a target CPU, the variables target_cpu and target_cpu_only
#   are available and are analogous to target_os and target_os_only.
```

```
# Specifying a target OS
#   An optional key named "target_os" may be added to a gclient file to specify
#   one or more additional operating systems that should be considered when
#   processing the deps_os/hooks_os dict of a DEPS file.
# ...
#   If the "target_os_only" key is also present and true, then *only* the
#   operating systems listed in "target_os" will be used.
```

So I believe `target_cpu` passed above only specifies _additional_ architectures. We're not setting `target_cpu_only`.

I just tested this by running `npm run init -- --target_os=linux --target_arch=arm64` with the changes in this PR. It added `target_cpu = ["arm64"]` to `.gclient` and installed the sysroot images for `x64`, `x86` and `arm64`. I take this to mean that it ran the hooks for all those architectures.

Reply:
but this is still going to be confusing for people trying to build arm64 because we normally don't have to change `.gclient` to do that so I think we should always include `arm64` if the host os is linux

Reply:
In my view, the fix in this PR is a simple, general solution to the problem we are experiencing, and other potential similar problems in the future. If we add a comment that explains why we're doing it this way, then I do not see how it is confusing. We're using `target_cpu` as it is intended, to define additional architectures. Furthermore, by not including `arm64` when it isn't needed, we are saving some time in the `npm run init` call, which is useful for CI and developers. I'm happy to make the change you suggested, of course, but for the reasons I just described my preference would be to add a comment to the existing implementation.

Reply:
yes, but `target_cpu` is not required for any other platform afaik and that's why this is going to cause problems and confusion for people. I can build x64 or arm64 on macos without this and same for ios/android (not sure about win since that's new). It also shouldn't increase CI times because there's no reason why CI can't run both archs from the same checkout (not sure if that happens now @mihaiplesa ?). How much additional time does it take?

Reply:
> this is going to cause problems

Can I ask out of curiosity why you think it will cause problems, and which ones?

> How much additional time does it take?

It takes 4-5 seconds on my machine.

Reply:
I already explained that the problem is that people building this locally are going to be confused because this extra step is not required for any other platform to build a different arch. Saving 4-5 seconds in sync is not worth the extra headache and confusion this is going to cause.

Reply:
Ok, I implemented it this way now in 17d0121c2c6b4f99d1eb9415e0057f6cad8f9e6c. Let me know what you think please @bridiver?